source/plugins/go/src/extension/extension.go

```diff
@@ -46,45 +54,49 @@ func GetInstance(flbLogger *log.Logger, containertype string) *Extension {
 	return singleton
 }
 
-func getExtensionData() (TaggedData, error) {
+func getExtensionData(extensionName string, extensionVersion string) (TaggedData, error) {
 	guid := uuid.New()
 	var extensionData TaggedData
-	taggedData := map[string]interface{}{"Request": "AgentTaggedData", "RequestId": guid.String(), "Tag": "ContainerInsights", "Version": "1"}
+	taggedData := map[string]interface{}{"Request": "AgentTaggedData", "RequestId": guid.String(), "Tag": extensionName, "Version": extensionVersion}
 	jsonBytes, err := json.Marshal(taggedData)
 	if err != nil {
-		logger.Printf("Error::mdsd/ama::Failed to marshal taggedData data. Error message: %s", string(err.Error()))
+		logger.Printf("Error::mdsd/ama::Failed to marshal taggedData data for extension: %s, error: %s", extensionName, string(err.Error()))
 		return extensionData, err
 	}
 
 	responseBytes, err := getExtensionConfigResponse(jsonBytes)
 	if err != nil {
-		logger.Printf("Error::mdsd/ama::Failed to get config response data. Error message: %s", string(err.Error()))
+		logger.Printf("Error::mdsd/ama::Failed to get config response data for extension: %s, error: %s", extensionName, string(err.Error()))
 		return extensionData, err
 	}
 	var responseObject AgentTaggedDataResponse
 	err = json.Unmarshal(responseBytes, &responseObject)
 	if err != nil {
-		logger.Printf("Error::mdsd/ama::Failed to unmarshal config response data. Error message: %s", string(err.Error()))
+		logger.Printf("Error::mdsd/ama::Failed to unmarshal config response data for extension: %s, error: %s", extensionName, string(err.Error()))
 		return extensionData, err
 	}
 
 	err = json.Unmarshal([]byte(responseObject.TaggedData), &extensionData)
+	if err != nil {
+		logger.Printf("Error::mdsd/ama::Failed to unmarshal config response TaggedData for extension: %s, error: %s", extensionName, string(err.Error()))
+		return extensionData, err
+	}
 
 	return extensionData, err
 }
 
-func getExtensionConfigs() ([]ExtensionConfig, error) {
-	extensionData, err := getExtensionData()
+func getExtensionConfigs(extensionName string, extensionVersion string) ([]ExtensionConfig, error) {
+	extensionData, err := getExtensionData(extensionName, extensionVersion)
 	if err != nil {
 		return nil, err
 	}
 	return extensionData.ExtensionConfigs, nil

```

Comment:
any validation needs to be done here before return it?

Reply:
we are good on this since getExtensionData handles the validation.